xbmc/utils/FileUtils.cpp

```diff
@@ -201,3 +203,36 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   }
   return dateAdded;
 }
+
+bool CFileUtils::CheckFileAccessAllowed(const std::string &filePath)
+{
+  std::string decodePath = CURL::Decode(filePath);
+  // deny access to security/privacy sensitive files
+  std::vector<std::string> noAccess = {"passwords.xml", "sources.xml", ".ssh", "/etc"};
+  for (const auto &f : noAccess)
+  {
+    if (decodePath.find(f) != std::string::npos)
+    {
+      CLog::Log(LOGERROR,"%s access to %s denied",  __FUNCTION__, decodePath.c_str());
+      return false;
+    }
+  }
+#if defined(TARGET_POSIX)
+  char *fullpath = realpath(filePath.c_str(), nullptr);
+  if (fullpath)
+  {
+    const std::string testpath = fullpath;
+    free(fullpath);
+    CLog::Log(LOGDEBUG,"%s filePath: %s decodePath: %s realpath: %s",  __FUNCTION__, filePath.c_str(), decodePath.c_str(), testpath.c_str());
+    // allow kodi paths
+    // TODO: most likely many more needed
+    std::string userHome = CSpecialProtocol::TranslatePath("special://home");
```

Comment:
make this a string list for easier extension.

Reply:
did you check the 2nd commit?

Reply:
aparently not!